patches/core/0016-adb-Use-a-default-_SC_GETPW_R_SIZE_MAX-size.patch

```diff
@@ -0,0 +1,31 @@
+From 7ee693cedb5c9d5e78fa730f65ef6eab2c0887fd Mon Sep 17 00:00:00 2001
+From: Robert Yang <decatf@gmail.com>
+Date: Wed, 24 Oct 2018 09:48:40 -0400
+Subject: [PATCH] adb: Use a default _SC_GETPW_R_SIZE_MAX size
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+sysconf(_SC_GETPW_R_SIZE_MAX) may return −1 if there is no hard limit on
+the the buffer size. Some libc implementations such as musl don't define
+this limit and will return -1.
+
+Use a default buffer size to handle this case.
+---
+ adb/adb_utils.cpp | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/adb/adb_utils.cpp b/adb/adb_utils.cpp
+index b236fb39f..6f7106f9d 100644
+--- a/adb/adb_utils.cpp
++++ b/adb/adb_utils.cpp
+@@ -249,6 +249,9 @@ std::string adb_get_homedir_path() {
+     struct passwd pwent;
+     struct passwd* result;
+     int pwent_max = sysconf(_SC_GETPW_R_SIZE_MAX);
++    if (pwent_max == -1) {
++        pwent_max = 16384;
```

Comment:
Makes sense.

As an extra note: I had submitted the patch to AOSP and it was merged already. If that patch makes it into a future android tag release (android-x.y.z) then it might be 16384 in that code.